### PR TITLE
PostgreSQL Flexible server added support for new api version 2024-08-01

### PR DESCRIPTION
**Pull request by nasc17**
This release uses GA api version 2024-08-01 for PostgreSQL flexible server.

**Features introduce by new stable version:**
Storage auto growth
IOPS scaling
Backup - Long Term Retention
Backup - On-demand
Geo-redundant backup encryption key - Revive Dropped
Server Logs
Migrations
Migration Pre-validation
Migration Roles
Private endpoint Migration
Private Endpoints
Read replicas - Switchover
Read replicas - Virtual Endpoints
Azure Defender / Threat Protection APIs
PG 16 support
PremiumV2_LRS storage type support
Location capabilities updates

### CODE REVIEW

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,29 @@
 # Release History
 
+## 1.2.0 (2023-10-25)
```
Comment (archerzz) [inline sdk/postgresql/Azure.ResourceManager.PostgreSql/CHANGELOG.md:3]: Need to specify a future date. Next time, please reserve some buffer time for review/modification. Thanks.

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/PostgreSqlFlexibleServersPrivateLinkResourceData.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        {
+            GroupId = groupId;
+            RequiredMembers = requiredMembers;
+            RequiredZoneNames = requiredZoneNames;
+            _serializedAdditionalRawData = serializedAdditionalRawData;
+        }
+
+        /// <summary> The private link resource group id. </summary>
+        [WirePath("properties.groupId")]
+        public string GroupId { get; }
```
Comment (archerzz) [inline sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/PostgreSqlFlexibleServersPrivateLinkResourceData.cs:79]: What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/maps/Azure.ResourceManager.Maps/src/autorest.md?plain=1#L25).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/search/Azure.Search.Documents/src/autorest.md?plain=1#L106-L115).

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/DbMigrationStatus.cs [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        public string MigrationOperation { get; }
+        /// <summary> Start date-time of a migration state. </summary>
+        [WirePath("startedOn")]
+        public DateTimeOffset? StartedOn { get; }
+        /// <summary> End date-time of a migration state. </summary>
+        [WirePath("endedOn")]
+        public DateTimeOffset? EndedOn { get; }
+        /// <summary> Number of tables queued for the migration of a DB. </summary>
+        [WirePath("fullLoadQueuedTables")]
+        public int? FullLoadQueuedTables { get; }
```
Comment (archerzz) [inline sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/DbMigrationStatus.cs:109]: For numbers, better to rename to `NumOfFullLoadQueuedTables`, or `CountOfFullLoadQueuedTables`, etc.

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/Replica.cs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+
+namespace Azure.ResourceManager.PostgreSql.FlexibleServers.Models
+{
+    /// <summary> Replica properties of a server. </summary>
+    public partial class Replica
```
Comment (archerzz) [inline sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/Replica.cs:14]: The name is too generic. Consider appending RP prefix or renaming it.
- use `prepend-rp-prefix` to automatically append PR name. See example [here](https://github.com/Azure/azure-sdk-for-net/blob/176fd476588e8e61ebce747d66714cb39947cae5/sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md?plain=1#L52-L57).
- use `rename-mapping` to rename a model. See example [here](https://github.com/Azure/azure-sdk-for-net/blob/176fd476588e8e61ebce747d66714cb39947cae5/sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md?plain=1#L58-L76)

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/ReplicationState.cs [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.PostgreSql.FlexibleServers.Models
+{
+    /// <summary> Gets the replication state of a replica server. This property is returned only for replicas api call. Supported values are Active, Catchup, Provisioning, Updating, Broken, Reconfiguring. </summary>
+    public readonly partial struct ReplicationState : IEquatable<ReplicationState>
```
Comment (archerzz) [inline sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/ReplicationState.cs:14]: Generic name

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/StorageType.cs [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.PostgreSql.FlexibleServers.Models
+{
+    /// <summary> Storage type for the server. Allowed values are Premium_LRS and PremiumV2_LRS, and default is Premium_LRS if not specified. </summary>
+    public readonly partial struct StorageType : IEquatable<StorageType>
```
Comment (archerzz) [inline sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/StorageType.cs:14]: Generic name

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/SslMode.cs [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.PostgreSql.FlexibleServers.Models
+{
+    /// <summary> Supported SSL modes for migration. VerifyFull is the recommended SSL mode for Single server migration. Prefer, Require are recommended SSL modes for other source types. </summary>
+    public readonly partial struct SslMode : IEquatable<SslMode>
```
Comment (archerzz) [inline sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/SslMode.cs:14]: Generic name

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/SourceType.cs [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.PostgreSql.FlexibleServers.Models
+{
+    /// <summary> Migration source server type : OnPremises, AWS, GCP, AzureVM, PostgreSQLSingleServer, AWS_RDS, AWS_AURORA, AWS_EC2, GCP_CloudSQL, GCP_AlloyDB, GCP_Compute, or EDB. </summary>
+    public readonly partial struct SourceType : IEquatable<SourceType>
```
Comment (archerzz) [inline sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/SourceType.cs:14]: Generic name

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/ServerSku.cs [hunk 9]
```diff
@@ -46,38 +46,26 @@ public partial class ServerSku
         private IDictionary<string, BinaryData> _serializedAdditionalRawData;
 
         /// <summary> Initializes a new instance of <see cref="ServerSku"/>. </summary>
-        /// <param name="name"> The name of the sku, typically, tier + family + cores, e.g. Standard_D4s_v3. </param>
-        /// <param name="tier"> The tier of the particular SKU, e.g. Burstable. </param>
-        /// <exception cref="ArgumentNullException"> <paramref name="name"/> is null. </exception>
-        internal ServerSku(string name, PostgreSqlFlexibleServerSkuTier tier)
```
Comment (archerzz) [inline sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/ServerSku.cs:52]: This is also a generic name. But if it's already published in an official release, then let's keep it as-is.

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/ValidationDetails.cs [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+
+namespace Azure.ResourceManager.PostgreSql.FlexibleServers.Models
+{
+    /// <summary> Details for the validation for migration. </summary>
+    public partial class ValidationDetails
```
Comment (archerzz) [inline sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/ValidationDetails.cs:14]: Generic name. Same as `ValidationMessage`, `ValidationState`.

**Review by archerzz — CHANGES_REQUESTED** [review]

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/PostgreSqlFlexibleServersPrivateLinkResourceData.cs (hunk 2 above)
Comment (nasc17) [reply sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/PostgreSqlFlexibleServersPrivateLinkResourceData.cs:79]: This is just a string. Example: "groupId": "batchAccount",

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/Replica.cs (hunk 4 above)
Comment (nasc17) [reply sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/Replica.cs]: Did prepend-rp-prefix

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/CHANGELOG.md (hunk 1 above)
Comment (nasc17) [reply sdk/postgresql/Azure.ResourceManager.PostgreSql/CHANGELOG.md]: Updated to November ship date :-)

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/ServerSku.cs (hunk 9 above)
Comment (nasc17) [reply sdk/postgresql/Azure.ResourceManager.PostgreSql/src/PostgreSqlFlexibleServers/Generated/Models/ServerSku.cs:52]: It was first published last preview release, updated name for this official release.

**Review by ArthurMa1978 — APPROVED** [review]

**Comment by nasc17** [comment]
@archerzz All requested changes have been addressed

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/CHANGELOG.md [hunk 11]
```diff
@@ -1,22 +1,37 @@
 # Release History
 
-## 1.2.0-beta.7 (Unreleased)
+## 1.2.0 (2024-11-05)
```
Comment (ambrahma) [inline sdk/postgresql/Azure.ResourceManager.PostgreSql/CHANGELOG.md:3]: is this referring to release date?

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/assets.json [hunk 12]
```diff
@@ -2,5 +2,5 @@
   "AssetsRepo": "Azure/azure-sdk-assets",
   "AssetsRepoPrefixPath": "net",
   "TagPrefix": "net/postgresql/Azure.ResourceManager.PostgreSql",
-  "Tag": "net/postgresql/Azure.ResourceManager.PostgreSql_3058e2076a"
+  "Tag": "net/postgresql/Azure.ResourceManager.PostgreSql_1f4154b2f1"
```
Comment (ambrahma) [inline sdk/postgresql/Azure.ResourceManager.PostgreSql/assets.json:5]: what is this Tag for? How is it different from TagPrefix above?

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/samples/Generated/Samples/Sample_PostgreSqlFlexibleServerBackupResource.cs [hunk 13]
```diff
@@ -14,12 +14,73 @@ namespace Azure.ResourceManager.PostgreSql.FlexibleServers.Samples
 {
     public partial class Sample_PostgreSqlFlexibleServerBackupResource
     {
+        // Create a new Backup for a flexible server
+        [NUnit.Framework.Test]
+        [NUnit.Framework.Ignore("Only verifying that the sample builds")]
```
Comment (ambrahma) [inline sdk/postgresql/Azure.ResourceManager.PostgreSql/samples/Generated/Samples/Sample_PostgreSqlFlexibleServerBackupResource.cs:19]: Are we ignoring all tests?

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/CHANGELOG.md (hunk 11 above)
Comment (nasc17) [reply sdk/postgresql/Azure.ResourceManager.PostgreSql/CHANGELOG.md:3]: Yes it needed to be a future date, I updated to November Ship Date.

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/assets.json (hunk 12 above)
Comment (nasc17) [reply sdk/postgresql/Azure.ResourceManager.PostgreSql/assets.json:5]: Previous tag referenced old recordings. New tag was created once pushing new recordings to asset repo.

[test guide](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/resourcemanager/Azure.ResourceManager/docs/TestGuide.md#test-proxy)

File: sdk/postgresql/Azure.ResourceManager.PostgreSql/samples/Generated/Samples/Sample_PostgreSqlFlexibleServerBackupResource.cs (hunk 13 above)
Comment (nasc17) [reply sdk/postgresql/Azure.ResourceManager.PostgreSql/samples/Generated/Samples/Sample_PostgreSqlFlexibleServerBackupResource.cs:19]: These are [samples](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/resourcemanager/Azure.ResourceManager/docs/TestGuide.md#samples) they get generated as unit tests. 

Our tests are found in [PostgreSqlFlexibleServerTests.cs](https://github.com/Azure/azure-sdk-for-net/blob/54cbfebc9ba94be3897fa31207efc568f1a636d3/sdk/postgresql/Azure.ResourceManager.PostgreSql/tests/Scenario/PostgreSqlFlexibleServerTests.cs)

**Review by archerzz — APPROVED** [review]

**Pull request merged**